src/Homebrew.jl

```diff
@@ -108,26 +131,47 @@ function install_brew()
     end
 end
 
-function update()
-    Git.run(`fetch origin`, dir=brew_prefix)
-    Git.run(`reset --hard origin/$BREW_BRANCH`, dir=brew_prefix)
-
-    # Find all namespaces inside <prefix>/Library/Taps, then search for taps
-    tapsdir = joinpath(brew_prefix,"Library","Taps")
-    namespaces = readdir(tapsdir)
-    ns_taps = [[joinpath(tapsdir, ns, tap) for tap in readdir(joinpath(tapsdir, ns))] for ns in namespaces]
-    taps = vcat(ns_taps...)
-
-    # Update each tap, one after another
-    for tap in taps
-        println("Updating tap $(basename(tap))")
-        Git.run(`fetch origin`, dir=tap)
-        TAP_BRANCH = Git.readchomp(`rev-parse --abbrev-ref HEAD`, dir=tap)
-        Git.run(`reset --hard origin/$TAP_BRANCH`, dir=tap)
-    end
+if VERSION < v"0.5.0-dev+522"
+    function update()
+        Git.run(`fetch origin`, dir=brew_prefix)
+        Git.run(`reset --hard origin/$BREW_BRANCH`, dir=brew_prefix)
+
+        # Find all namespaces inside <prefix>/Library/Taps, then search for taps
+        tapsdir = joinpath(brew_prefix,"Library","Taps")
+        namespaces = readdir(tapsdir)
+        ns_taps = [[joinpath(tapsdir, ns, tap) for tap in readdir(joinpath(tapsdir, ns))] for ns in namespaces]
+        taps = vcat(ns_taps...)
+
+        # Update each tap, one after another
+        for tap in taps
+            println("Updating tap $(basename(tap))")
+            Git.run(`fetch origin`, dir=tap)
+            TAP_BRANCH = Git.readchomp(`rev-parse --abbrev-ref HEAD`, dir=tap)
+            Git.run(`reset --hard origin/$TAP_BRANCH`, dir=tap)
+        end
 
-    # Finally, upgrade outdated packages.
-    upgrade()
+        # Finally, upgrade outdated packages.
+        upgrade()
+    end
+else
+    function update()
+        repo = LibGit2.GitRepo(brew_prefix)
+        remote = LibGit2.get(LibGit2.GitRemote, repo, "origin")
+        
+        tapsdir = joinpath(brew_prefix,"Library","Taps")
+        namespaces = readdir(tapsdir)
+        ns_taps = [[joinpath(tapsdir, ns, tap) for tap in readdir(joinpath(tapsdir, ns))] for ns in namespaces]
+        taps = vcat(ns_taps...)
+
+        # Update each tap, one after another
+        for tap in taps
+            println("Updating tap $(basename(tap))")
+            LibGit2.fetch(remote,[BREW_BRANCH])
```

Comment:
Are you positive git is installed by default even when Xcode and command-line tools aren't? We don't directly run the homebrew install script here, right? How good are you at reading ruby, what happens here if it can't find anything? https://github.com/Homebrew/install/blob/9c46b270eb69631d692c009ecba3d33ddfe6d480/install#L86